src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -423,7 +424,11 @@ private ValueTask SendFrameLockAcquiredNonCancelableAsync(MessageOpcode opcode,
                 // the task, and we're done.
                 if (writeTask.IsCompleted)
                 {
-                    return writeTask;
+                    flushTask = _stream.FlushAsync();
+                    if (flushTask.IsCompleted)
```

Comment:
Like this, in case of `flushTask.IsCompleted == false` you will end up throwing away this task and calling Flush again inside `WaitForWriteTaskAsync`.

I feel that while for code readability it might be better to just have `writeTask` to contain two instructions (Write and then Flush) -- but introducing additional task might regress the perf here. So maybe it is better to pass `flushTask` to WaitForWriteTaskAsync as well.

Reply:
The simplest fix would be to add:
```
writeTask.GetAwaiter().GetResult();
writeTask = stream.FlushAsync(...);
```

Reply:
Actually, instead of:
writeTask.GetAwaiter().GetResult()
it'd be better to do:
if (! writeTask.IsCompletedSuccessfully) return writeTask;

Reply:
But writeTask would still have to be awaited in successful case? because it's ValueTask

Reply:
ah, it's a value task. then I'd suggest doing both of the lines

Reply:
Thanks, I added the first option